Apache image can't utilize ldapauth addon because container lacks `libldap2-dev`

`friendica.log` renders:
```
2019-11-12 10:25:08 index [INFO]: ldapauth: not configured or missing php-ldap module [] - {"file":"ldapauth.php","line":105,"function":"ldapauth_authenticate","uid":"3c55bf","process_id":39}
```
for example, after my trying to fill out and run an `addon.config.php` with an `ldapauth` block.

actually, it looks like it might be `libldap2-dev` that's missing, because 
```
root@4a678cd31a50:/var/www/html# docker-php-ext-configure ldap --with-libdir=lib/x86_64-linux-gnu/
```
fails during `configure` on
```
configure: error: Cannot find ldap.h
```

but succeeds when `libldap2-dev` is present. I'm still trying to work out if it's both the `php7.3-ldap` and `libldap2-dev` packages are required, or just the latter. stay tuned.

--- a/2019.09/apache/Dockerfile
+++ b/2019.09/apache/Dockerfile
@@ -36,6 +36,9 @@ RUN set -ex; \
         libzip-dev \
         libldap2-dev \
     ; \
+    \
+        debMultiarch="$(dpkg-architecture --query DEB_BUILD_MULTIARCH)"; \
+    \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
@@ -43,8 +46,8 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     docker-php-ext-configure ldap \
-    	--with-libdir=lib/x86_64-linux-gnu/ \
-    ; \
+        	--with-libdir=lib/$debMultiarch/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
         gd \

--- a/2019.09/apache/Dockerfile
+++ b/2019.09/apache/Dockerfile
@@ -34,12 +34,16 @@ RUN set -ex; \
         libfreetype6-dev \
         librsvg2-2 \
         libzip-dev \
+        libldap2-dev \
     ; \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
         --with-png-dir=/usr/include/ \
         --with-jpeg-dir=/usr/include/ \
+    ; \
+    docker-php-ext-configure ldap \
+    	--with-libdir=lib/x86_64-linux-gnu/ \
     ; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
@@ -48,12 +52,13 @@ RUN set -ex; \
         opcache \
         ctype \
         pcntl \
+        ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install apcu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install apcu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/2019.09/fpm-alpine/Dockerfile
+++ b/2019.09/fpm-alpine/Dockerfile
@@ -28,6 +28,7 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
+        libldap2-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -37,18 +38,22 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
+    docker-php-ext-configure ldap \
+        	--with-libdir=lib/x86_64-linux-gnu/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \
        zip \
        opcache \
        pcntl \
+       ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install APCu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install APCu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/2019.09/fpm-alpine/Dockerfile
+++ b/2019.09/fpm-alpine/Dockerfile
@@ -28,7 +28,8 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
-        libldap2-dev \
+        icu-dev \
+        openldap-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -38,9 +39,6 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
-    docker-php-ext-configure ldap \
-        	--with-libdir=lib/x86_64-linux-gnu/ \
-	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \

--- a/2019.09/fpm/Dockerfile
+++ b/2019.09/fpm/Dockerfile
@@ -36,6 +36,9 @@ RUN set -ex; \
         libzip-dev \
         libldap2-dev \
     ; \
+    \
+        debMultiarch="$(dpkg-architecture --query DEB_BUILD_MULTIARCH)"; \
+    \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
@@ -43,8 +46,8 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     docker-php-ext-configure ldap \
-    	--with-libdir=lib/x86_64-linux-gnu/ \
-    ; \
+        	--with-libdir=lib/$debMultiarch/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
         gd \

--- a/2019.09/fpm/Dockerfile
+++ b/2019.09/fpm/Dockerfile
@@ -34,12 +34,16 @@ RUN set -ex; \
         libfreetype6-dev \
         librsvg2-2 \
         libzip-dev \
+        libldap2-dev \
     ; \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
         --with-png-dir=/usr/include/ \
         --with-jpeg-dir=/usr/include/ \
+    ; \
+    docker-php-ext-configure ldap \
+    	--with-libdir=lib/x86_64-linux-gnu/ \
     ; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
@@ -48,12 +52,13 @@ RUN set -ex; \
         opcache \
         ctype \
         pcntl \
+        ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install apcu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install apcu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/2019.12-dev/apache/Dockerfile
+++ b/2019.12-dev/apache/Dockerfile
@@ -36,6 +36,9 @@ RUN set -ex; \
         libzip-dev \
         libldap2-dev \
     ; \
+    \
+        debMultiarch="$(dpkg-architecture --query DEB_BUILD_MULTIARCH)"; \
+    \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
@@ -43,8 +46,8 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     docker-php-ext-configure ldap \
-    	--with-libdir=lib/x86_64-linux-gnu/ \
-    ; \
+        	--with-libdir=lib/$debMultiarch/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
         gd \

--- a/2019.12-dev/apache/Dockerfile
+++ b/2019.12-dev/apache/Dockerfile
@@ -34,12 +34,16 @@ RUN set -ex; \
         libfreetype6-dev \
         librsvg2-2 \
         libzip-dev \
+        libldap2-dev \
     ; \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
         --with-png-dir=/usr/include/ \
         --with-jpeg-dir=/usr/include/ \
+    ; \
+    docker-php-ext-configure ldap \
+    	--with-libdir=lib/x86_64-linux-gnu/ \
     ; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
@@ -48,12 +52,13 @@ RUN set -ex; \
         opcache \
         ctype \
         pcntl \
+        ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install apcu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install apcu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/2019.12-dev/fpm-alpine/Dockerfile
+++ b/2019.12-dev/fpm-alpine/Dockerfile
@@ -28,6 +28,7 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
+        libldap2-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -37,18 +38,22 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
+    docker-php-ext-configure ldap \
+        	--with-libdir=lib/x86_64-linux-gnu/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \
        zip \
        opcache \
        pcntl \
+       ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install APCu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install APCu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/2019.12-dev/fpm-alpine/Dockerfile
+++ b/2019.12-dev/fpm-alpine/Dockerfile
@@ -28,7 +28,8 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
-        libldap2-dev \
+        icu-dev \
+        openldap-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -38,9 +39,6 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
-    docker-php-ext-configure ldap \
-        	--with-libdir=lib/x86_64-linux-gnu/ \
-	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \

--- a/2019.12-dev/fpm/Dockerfile
+++ b/2019.12-dev/fpm/Dockerfile
@@ -36,6 +36,9 @@ RUN set -ex; \
         libzip-dev \
         libldap2-dev \
     ; \
+    \
+        debMultiarch="$(dpkg-architecture --query DEB_BUILD_MULTIARCH)"; \
+    \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
@@ -43,8 +46,8 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     docker-php-ext-configure ldap \
-    	--with-libdir=lib/x86_64-linux-gnu/ \
-    ; \
+        	--with-libdir=lib/$debMultiarch/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
         gd \

--- a/2019.12-dev/fpm/Dockerfile
+++ b/2019.12-dev/fpm/Dockerfile
@@ -34,12 +34,16 @@ RUN set -ex; \
         libfreetype6-dev \
         librsvg2-2 \
         libzip-dev \
+        libldap2-dev \
     ; \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
         --with-png-dir=/usr/include/ \
         --with-jpeg-dir=/usr/include/ \
+    ; \
+    docker-php-ext-configure ldap \
+    	--with-libdir=lib/x86_64-linux-gnu/ \
     ; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
@@ -48,12 +52,13 @@ RUN set -ex; \
         opcache \
         ctype \
         pcntl \
+        ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately
-    pecl install apcu-5.1.17; \
-    pecl install memcached-3.1.3; \
-    pecl install redis-5.0.2; \
+    pecl install apcu-5.1.18; \
+    pecl install memcached-3.1.4; \
+    pecl install redis-5.1.1; \
     pecl install imagick-3.4.4; \
     \
     docker-php-ext-enable \

--- a/Dockerfile-alpine.template
+++ b/Dockerfile-alpine.template
@@ -27,6 +27,7 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
+        libldap2-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -36,12 +37,16 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
+    docker-php-ext-configure ldap \
+        	--with-libdir=lib/x86_64-linux-gnu/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \
        zip \
        opcache \
        pcntl \
+       ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately

--- a/Dockerfile-alpine.template
+++ b/Dockerfile-alpine.template
@@ -27,7 +27,8 @@ RUN set -ex; \
         librsvg \
         pcre-dev \
         libzip-dev \
-        libldap2-dev \
+        icu-dev \
+        openldap-dev \
     ; \
     \
     docker-php-ext-configure gd \
@@ -37,9 +38,6 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     \
-    docker-php-ext-configure ldap \
-        	--with-libdir=lib/x86_64-linux-gnu/ \
-	; \
     docker-php-ext-install -j "$(nproc)" \
        pdo_mysql \
        gd \

--- a/Dockerfile-debian.template
+++ b/Dockerfile-debian.template
@@ -33,12 +33,16 @@ RUN set -ex; \
         libfreetype6-dev \
         librsvg2-2 \
         libzip-dev \
+        libldap2-dev \
     ; \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
         --with-png-dir=/usr/include/ \
         --with-jpeg-dir=/usr/include/ \
+    ; \
+    docker-php-ext-configure ldap \
+    	--with-libdir=lib/x86_64-linux-gnu/ \
     ; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
@@ -47,6 +51,7 @@ RUN set -ex; \
         opcache \
         ctype \
         pcntl \
+        ldap \
     ; \
     \
 # pecl will claim success even if one install fails, so we need to perform each install separately

--- a/Dockerfile-debian.template
+++ b/Dockerfile-debian.template
@@ -35,6 +35,9 @@ RUN set -ex; \
         libzip-dev \
         libldap2-dev \
     ; \
+    \
+        debMultiarch="$(dpkg-architecture --query DEB_BUILD_MULTIARCH)"; \
+    \
     docker-php-ext-configure gd \
         --with-gd \
         --with-freetype-dir=/usr/include/ \
@@ -42,8 +45,8 @@ RUN set -ex; \
         --with-jpeg-dir=/usr/include/ \
     ; \
     docker-php-ext-configure ldap \
-    	--with-libdir=lib/x86_64-linux-gnu/ \
-    ; \
+        	--with-libdir=lib/$debMultiarch/ \
+	; \
     docker-php-ext-install -j "$(nproc)" \
         pdo_mysql \
         gd \

